django/db/models/expressions.py

```diff
@@ -485,13 +485,17 @@ def select_format(self, compiler, sql, params):
 class Expression(BaseExpression, Combinable):
     """An expression that can be combined with other expressions."""
 
+    @classproperty
+    @functools.lru_cache(maxsize=128)
+    def _constructor_signature(cls):
+        return inspect.signature(cls.__init__)
+
     @cached_property
     def identity(self):
-        constructor_signature = inspect.signature(self.__init__)
         args, kwargs = self._constructor_args
-        signature = constructor_signature.bind_partial(*args, **kwargs)
+        signature = self._constructor_signature.bind_partial(self, *args, **kwargs)
         signature.apply_defaults()
-        arguments = signature.arguments.items()
+        arguments = list(signature.arguments.items())[1:]
```

Comment:
An alternative here would be

```suggestion
        arguments = iter(signature.arguments.items())
        next(arguments)
```

To avoid a temporary list creation and use the `dict_itemiterator` that would get created anyway in the loop below.

Reply:
I think it's worth doing :+1: